Core/GDCore/IDE/Events/EventsVariableReplacer.cpp

```diff
@@ -364,6 +365,16 @@ bool EventsVariableReplacer::DoVisitInstruction(gd::Instruction& instruction,
                 parameterIndex, ExpressionParser2NodePrinter::PrintNode(*node));
           }
         }
+
+        if (gd::ParameterMetadata::IsExpression("variable", type)) {
+          const auto &variableName = instruction.GetParameter(parameterIndex).GetPlainString();
```

Comment:
```suggestion
          const auto &variableName = parameterValue.GetPlainString();
```

Reply:
No, but I used a variable name to make it clear that it's not the same as `parameterValue`.